### PR TITLE
Start recording konflux builds 

### PR DESCRIPTION
**Pull request by thegreyd**
Changes
- Start recording konflux build at trigger time and complete time
- Some log message updates, post UI link and pipelinerun "message"
- Also record some extra info that is available for failed brew builds, like nvr from rebase time
- In elliott find-k-builds, add flags to search by engine/outcome and output to json

Test:

```
$ ./doozer --assembly test -g openshift-4.18 -i openshift-base-rhel9 --latest-parent-version
beta:images:konflux:build --konflux-kubeconfig ~/konflux.kubeconfig.p02 --konflux-namespace ocp-art-tenant
...
2024-10-01 18:33:33,719 doozerlib.backend.konflux_image_builder INFO [openshift-base-rhel9] Created PipelineRun:
 https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/workspaces/ocp-art/applications/
openshift-4-18/pipelineruns/doozer-build-openshift-4-18-openshift-base-rhel9-84vv4
2024-10-01 18:33:33,719 artcommonlib.bigquery INFO Bound to table openshift-art.events.builds
2024-10-01 18:33:36,115 doozerlib.backend.konflux_image_builder INFO Konflux build info stored successfully
2024-10-01 18:33:36,115 doozerlib.backend.konflux_image_builder INFO [openshift-base-rhel9] Waiting for PipelineRun 
doozer-build-openshift-4-18-openshift-base-rhel9-84vv4 to complete...
2024-10-01 18:33:39,408 doozerlib.backend.konflux_image_builder INFO PipelineRun doozer-build-openshift-4-18-
openshift-base-rhel9-84vv4 status: Unknown PipelineRun ocp-art-tenant/doozer-build-openshift-4-18-openshift-base-
rhel9-84vv4 awaiting remote resource
2024-10-01 18:33:40,116 doozerlib.backend.konflux_image_builder INFO PipelineRun doozer-build-openshift-4-18-
openshift-base-rhel9-84vv4 status: Unknown Tasks Completed: 0 (Failed: 0, Cancelled 0), Incomplete: 14, Skipped: 0
```

```
$ ./elliott --assembly test -g openshift-4.18 -i openshift-base-rhel9 find-k-builds 
-k image -e konflux --outcome pending --output json
...
2024-10-01 18:53:28,901 artcommonlib.bigquery INFO Bound to table openshift-art.events.builds
2024-10-01 18:53:28,901 art_tools.elliottlib.cli.find_konflux_builds_cli INFO searching for konflux builds of kind=image 
in group=openshift-4.18 with outcome=pending
2024-10-01 18:53:28,901 artcommonlib.konflux.konflux_db INFO Searching for openshift-base-rhel9 builds completed
 before 2024-10-01 18:53:28.901454
2024-10-01 18:53:30,046 art_tools.elliottlib.cli.find_konflux_builds_cli INFO Found 1 builds
{
    "name": "openshift-base-rhel9",
    "group": "openshift-4.18",
    "version": "v4.18.0",
    "release": "202410010953.p0.gb45ea65.assembly.test.el9",
    "assembly": "test",
    "source_repo": "https://github.com/openshift-eng/ocp-build-data",
    "commitish": "b45ea65bf6606c558b1a18b92ad878f42a411894",
    "rebase_repo_url": "https://github.com/openshift-eng/ocp-build-data",
    "rebase_commitish": "36dc20cd4a3a976249a26fb3d30a10092f0a7770",
    "start_time": "2024-10-01T18:33:33",
    "end_time": "2024-10-01T18:33:33",
    "artifact_type": "image",
    "engine": "konflux",
    "image_pullspec": "",
    "image_tag": "",
    "outcome": "pending",
    "art_job_url": "n/a",
    "build_pipeline_url": "https://konflux.apps.stone-prod-p02.hjvn.p1.openshiftapps.com/application-pipeline/
workspaces/ocp-art/applications/openshift-4-18/pipelineruns/doozer-build-openshift-4-18-openshift-base-rhel9-84vv4",
    "pipeline_commit": "n/a",
    "schema_level": 1,
    "ingestion_time": "2024-10-01T18:33:33",
    "record_id": "4b1f4419-b596-d1ff-bd55-fdae231461a1",
    "build_id": "doozer-build-openshift-4-18-openshift-base-rhel9-84vv4",
    "nvr": "openshift-base-rhel9-container-v4.18.0-202410010953.p0.gb45ea65.assembly.test.el9",
    "el_target": "el9",
    "arches": [
        "x86_64",
        "ppc64le",
        "s390x",
        "aarch64"
    ],
    "installed_packages": [],
    "parent_images": [],
    "embargoed": false
}

```

### CODE REVIEW

**Review by locriandev — COMMENTED** [review]
/lgtm

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 1]
```diff
@@ -199,10 +210,110 @@ async def _start_build(self, metadata: ImageMetadata, build_repo: BuildRepo, dyn
...
+
+            build_record_params = {
+                'name': metadata.distgit_key,
+                'version': version,
+                'release': release,
+                'el_target': f'el{isolate_el_version_in_release(release)}',
+                'arches': metadata.get_arches(),
+                'embargoed': 'p1' in release.split('.'),
+                'start_time': datetime.now(),
+                'end_time': datetime.now(),
```
Comment (locriandev) [inline doozer/doozerlib/backend/konflux_image_builder.py:257]:
```suggestion
                'end_time': None,
```

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 2]
```diff
@@ -199,10 +210,110 @@ async def _start_build(self, metadata: ImageMetadata, build_repo: BuildRepo, dyn
...
+                    'start_time': datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S.%f'),
+                    'end_time': datetime.strptime(end_time, '%Y-%m-%d %H:%M:%S.%f'),
+                    'image_tag': image_digest.split('sha256:')[-1],
+                })
+            elif outcome == KonfluxBuildOutcome.FAILURE:
+                start_time = pipelinerun.status.startTime
+                end_time = pipelinerun.status.completionTime
+                build_record_params.update({
+                    'start_time': datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S.%f'),
+                    'end_time': datetime.strptime(end_time, '%Y-%m-%d %H:%M:%S.%f'),
```
Comment (locriandev) [inline doozer/doozerlib/backend/konflux_image_builder.py:307]:
```suggestion
                    'start_time': datetime.strptime(start_time, '%Y-%m-%dT%H:%M:%SZ'),
                    'end_time': datetime.strptime(end_time, '%Y-%m-%dT%H:%M:%SZ')
```

**Review by locriandev — CHANGES_REQUESTED** [review]

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 3]
```diff
@@ -199,10 +210,110 @@ async def _start_build(self, metadata: ImageMetadata, build_repo: BuildRepo, dyn
...
+
+            component_name = df.labels['com.redhat.component']
+            version = df.labels['version']
+            release = df.labels['release']
+            nvr = "-".join([component_name, version, release])
+
+            pipelinerun_name = pipelinerun['metadata']['name']
+            build_pipeline_url = self.build_pipeline_url(pipelinerun)
+
+            build_record_params = {
```
Comment (locriandev) [inline doozer/doozerlib/backend/konflux_image_builder.py:249]: I noticed we're not storing `installed_packages` and `parent_images`, which is crucial for the ocp4-scan implementation. Are we planning to add this at a later time?

Comment (thegreyd) [reply doozer/doozerlib/backend/konflux_image_builder.py:249]: I'm okay with merging this and quickly following up

File: doozer/doozerlib/backend/konflux_image_builder.py [hunk 4]
```diff
@@ -199,10 +210,110 @@ async def _start_build(self, metadata: ImageMetadata, build_repo: BuildRepo, dyn
...
+
+                start_time = pipelinerun.status.startTime
+                end_time = pipelinerun.status.completionTime
+
+                build_record_params.update({
+                    'image_pullspec': image_pullspec,
+                    'installed_packages': [],  # TODO: populate this
+                    'parent_images': [],  # TODO: populate this
+                    'start_time': datetime.strptime(start_time, '%Y-%m-%d %H:%M:%S.%f'),
+                    'end_time': datetime.strptime(end_time, '%Y-%m-%d %H:%M:%S.%f'),
```
Comment (thegreyd) [inline doozer/doozerlib/backend/konflux_image_builder.py:299]:
```suggestion
                    'start_time': datetime.strptime(start_time, '%Y-%m-%dT%H:%M:%SZ'),
                    'end_time': datetime.strptime(end_time, '%Y-%m-%dT%H:%M:%SZ'),
```

**Review by locriandev — APPROVED** [review]
/lgtm

**Pull request merged**